sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/ODataDataObject.scala

```diff
@@ -0,0 +1,863 @@
+  }
+
+  def writeToFile(fileName: String, content: String) : Unit = {
+    initTemporaryDirectory()
+    ioc.writeHadoopFile(ioc.newHadoopPath(temporaryTargetDirectoryPath, fileName), content, filesystem)
+  }
+
+  def generateFileName() : String = {
+    s"${this.getResponseCount}.json"
+  }
+
+  /**
+   * Adds the provided response to the buffer.
+   *
+   * @param response : The response to be added
+   */
+  override def addResponse(response: String): Unit = {
+
+    this.writeToFile(this.generateFileName(), response)
+
+    super.addResponse(response)
+  }
+
+  /**
+   * Creates and returns a DataFrame referencing the stored responses. Each response is one record.
+   *
+   * @return : The Dataframe referencing the stored responses
+   */
+  override def getDataFrame: DataFrame = {
+    val session = context.sparkSession
+    val dataFrame = session.read.option("wholetext", value = true).text(this.temporaryTargetDirectoryPath.toString).withColumnRenamed("value", "responseString")
+    dataFrame
+  }
+
+  /**
+   * Switches this buffer to next stage and copies all already buffered responses, to the new instance - if necessary
+   *
+   * @return The new response buffer instance.
+   */
+  override def switchIfNecessary(): ODataResponseBuffer = {
+    this
+  }
+}
+/**
+ * Factory object to create and evolve the response buffers
+ */
+
+
+/**
+ * [[DataObject]] of type OData.
+ *
+ * @param schema : Schema of the expected output in the form of <<array< struct< columnA:string, columnB: integer ... >>
+ * @param baseUrl : Base URL of the OData Service like https://xxx.crm4.dynamics.com/api/data/v9.2/
+ * @param tableName : Name of the table which needs to be accessed
+ * @param sourceFilters : Optional. OData filter string which will be applied to the access operation like "objecttypecode eq 'task' and createdon ge 2024-01-01T00:00:00.000Z"
+ * @param timeouts : Optional. Timeout settings of type [[HttpTimeoutConfig]]
+ * @param authorization: Optional. Authorization credentials of type [[ODataAuthorization]]
+ * @param changeDateColumnName: Optional. Name of the column which will be used to read incrementally. like "modifiedon"
+ * @param nRetry: Optional. Number of retries after a failed attempt, default = 1
+ * @param responseBufferSetup: Optional. Setup for response buffers of type [[ODataResponseBufferSetup]]
+ * @param maxRecordCount: Optional. Maximum number of records to be extracted.
+ * @param metadata
+ * @param instanceRegistry
+ */
+case class ODataDataObject(override val id: DataObjectId,
+                           schema: String,
+                           baseUrl : String,
+                           tableName: String,
+                           sourceFilters: Option[String] = None,
+                           timeouts: Option[HttpTimeoutConfig] = None,
+                           authorization: Option[ODataAuthorization] = None,
+                           changeDateColumnName: Option[String] = None,
```

Comment:
This is named incrementalOutputExpr in JdbcTableDataObject.

Reply:
I would not call it "incrementalOutputExpr" because its not an expression. It must be the name of a datetime column and the DataObject will treat it accordingly.